File: module_infrastructure/src/main/java/project/redis/screening/repository/ScreeningRepository.java
package project.redis.screening.repository;

import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;
import project.redis.movie.entity.MovieEntity;
import project.redis.screening.entity.ScreeningEntity;

public interface ScreeningRepository extends JpaRepository<ScreeningEntity, Long> {

    List<ScreeningEntity> findByMovie(MovieEntity movieEntity);
}

File: infrastructure/src/main/kotlin/com/hanghe/redis/mysql/screening/ScreeningRepository.kt
package com.hanghe.redis.mysql.screening

import com.hanghe.redis.screening.ScreeningEntity

interface ScreeningRepository {

    fun findByMovieIdOrderByStartTime(movieId: Long): List<ScreeningEntity>

    fun saveAll(movies: List<ScreeningEntity>): List<ScreeningEntity>
}
